lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -718,6 +713,43 @@ protected function prepareUpdateData($entity, bool $isInsert = false)
         return $result;
     }
 
+    /**
+     * Decides if an extra update is required for the entity being persisted
+     * This is only required if the associated entity is different from the current one,
+     * and if the current entity relies on the database to generate its id
+     *
+     * @param object      $entity
+     * @param object|null $newVal
+     *
+     * @return bool Returns true is an extra update is required
+     */
+    private function isExtraUpdateRequired($entity, $newVal): bool
+    {
+        if ($newVal === null) {
+            return false;
+        }
+
+        $oid = spl_object_id($newVal);
+        $uow = $this->em->getUnitOfWork();
+        if (! (isset($this->queuedInserts[$oid]) || $uow->isScheduledForInsert($newVal))) {
+            return false;
+        }
+
+        if ($newVal !== $entity) {
+            return true;
+        }
+
+        $identifiers = $this->class->getIdentifier();
+        // Only single-column identifiers are supported
+        $entityChangeSet = $uow->getEntityChangeSet($entity);
+        if (count($identifiers) === 1 && isset($entityChangeSet[$identifiers[0]])) {
+            // Extra update is required if the current entity does not have yet a value for its identifier
+            return $entityChangeSet[$identifiers[0]][1] === null;
+        }
+
+        return true;
```

Comment:
This execution branch not being reached here seems to indicate a missing composite key test

Reply:
Thank you for your insight 👍 
I changed the code to remove this case as I'm only targeting self-referencing entities for this optimization.
So it doesn't make sense to check if this is composite key